cmd/chainsimulator/main.go

```diff
@@ -395,16 +395,27 @@ func loadMainConfig(filepath string) (config.Config, error) {
 	return cfg, err
 }
 
-func determineOverrideConfigFiles(ctx *cli.Context) []string {
+func determineOverrideConfigFiles(ctx *cli.Context, isSovereign bool) []string {
 	overrideFiles := strings.Split(ctx.GlobalString(nodeOverrideConfigurationFile.Name), overrideConfigFilesSeparator)
 
+	if isSovereign {
+		overrideFiles = orderAndAppendDefaultOverrideFileIfNeeded(overrideFiles, nodeOverrideSovereignDefaultFilename)
+	}
+	return orderAndAppendDefaultOverrideFileIfNeeded(overrideFiles, nodeOverrideDefaultFilename)
+}
+
+// this function will put the default node override file first
+func orderAndAppendDefaultOverrideFileIfNeeded(overrideFiles []string, nodeOverrideDefaultFilename string) []string {
+	files := make([]string, 0)
+
 	for _, filename := range overrideFiles {
 		if strings.Contains(filename, nodeOverrideDefaultFilename) {
-			return overrideFiles
+			continue
```

Comment:
Are you sure we are not breaking any future development here for normal run type?
For example, what if we add a new file in the slice for normal run type? Will this still behave correctly?

Reply:
Should not break anything, I runned testing-suite with this branch and all works as expected. 
This function will make sure the default overrides are first, which is correct